membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -260,21 +257,25 @@ class AccountController(
           contact <- SimpleEitherT.fromFutureOption(services.contactRepository.get(user), s"No SF user $user")
           subscription <- SimpleEitherT(
             services.subscriptionService
-              .current[SubscriptionPlan.Contributor](contact)
-              .map(subs => subscriptionSelector(subscriptionNameOption, s"the sfUser $contact")(subs)),
+              .current[SubscriptionPlan.AnyPlan](contact)
+              .map(subs => subscriptionSelector(subscriptionName, s"the sfUser $contact", subs)),
           )
-          billingPeriod = subscription.plan.charges.billingPeriod.asInstanceOf[RecurringPeriod]
-          applyFromDate = subscription.plan.chargedThrough.getOrElse(subscription.plan.start)
-          currency = subscription.plan.charges.price.prices.head.currency
+          contributionPlan <- SimpleEitherT.fromEither(subscription.plan match {
+            case p: SubscriptionPlan.Contributor @unchecked /* extra guard needed due to type erasure --> */ if p.product == Contribution => Right(p)
```

Comment:
this is the key update - now we are asking for it as AnyPlan, we need to downcast it back to Contributor